gradle/javaModule.gradle

```diff
@@ -127,6 +127,13 @@ checkstyle {
 }
 
 tasks.withType(Test) {
+    // Junit5 parallel tests
+    systemProperties["junit.jupiter.execution.parallel.enabled"] = true
```

Comment:
Do we need to tune `-PtestForks` parameters in the Jenkinsfile, and check if this is okay for github actions?
`dev/docs/tests` also mentions the parameter.

E.g if I run `-PtestForks=10` with this change I ran into out-of-memory errors because it's duplicating the concurrency.

```
Caused by: java.lang.OutOfMemoryError: Java heap space
	at java.base/java.util.ArrayList.<init>(ArrayList.java:156)
	at io.crate.statistics.ColumnStatsTest.arbitraryParams(ColumnStatsTest.java:70)
	at java.base/java.lang.invoke.DirectMethodHandle$Holder.invokeVirtual(DirectMethodHandle$Holder)
	at java.base/java.lang.invoke.LambdaForm$MH/0x0000000801008000.invoke(LambdaForm$MH)
	at java.base/java.lang.invoke.Invokers$Holder.invokeExact_MT(Invokers$Holder)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invokeImpl(DirectMethodHandleAccessor.java:154)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
```

(Also, what's the advantage of this over the existing mechanism?)

Reply:
- The benefit is to be able to also run tests inside a class, in parallel, which can for sure drop the time for unit tests, and even for some integration tests (if they support this concurrency)
- I will test this, I'm usually running the tests on my machine with: `./gradlew clean test --parallel -PtestForks=18 -DforkEvery=20` so maybe not forking on every test prevented this OutOfMemory
- I will also test github actions, thx! I totally forgot about that

Reply:
For sure, there is absolutely no rush.

Reply:
I haven't managed to make it fail, I've even tried with `maxHeapSize = "1024m"` in the test configuration. Do you have any other special config yourslef? which gradle command did you use?